ci/ray_ci/upgrade_version.py

```diff
@@ -0,0 +1,96 @@
+import click
+import os
+import subprocess
+from typing import Set
+
+bazel_workspace_dir = os.environ.get("BUILD_WORKSPACE_DIRECTORY", "")
+
+non_java_files = {
+    "ci/ray_ci/utils.py",
+    "python/ray/_version.py",
+    "src/ray/common/constants.h",
+}
+
+DEFAULT_NON_JAVA_VERSION = "3.0.0.dev0"
+DEFAULT_JAVA_VERSION = "2.0.0-SNAPSHOT"
+
+
+def list_java_files(root_dir: str):
+    """
+    Scan the directories and return the set of pom.xml and pom_template.xml files.
+    """
+    files = set()
+    for current_root_dir, _, file_names in os.walk(root_dir):
+        for file_name in file_names:
+            if file_name in ["pom.xml", "pom_template.xml"]:
+                files.add(os.path.join(current_root_dir, file_name))
+    return files
+
+
+def get_check_output(file_path: str):
+    return subprocess.check_output(["python", file_path], text=True)
+
+
+def get_current_version(root_dir: str):
+    """
+    Scan for current Ray version and return the current versions.
+    """
+    version_file_path = os.path.join(root_dir, "python/ray/_version.py")
+    ray_version_output = get_check_output(version_file_path).split(" ")
```

Comment:
just use `split()` without args? so that it just splits with whitespaces?

Reply:
done